### PR TITLE
Code cleanups in stl/src

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* `awint.h` is `stl/src` internal, and is included by C++ translation units only (although this wasn't the case in the past). We can and should accordingly change `(void)` for zero-parameter functions to `()`, which was already being used for `__crtGetTickCount64()`.
* The function `__crtTerminateProcess()` is neither used nor exported, so it can be deleted. (Apparently it was exported by DLLs before `msvcp140.dll`.)
* Change `typedef` to `using`.
* In `awint.h`, mark function pointers as `const`. Use `auto` for style.
* In `awint.h`, remove unnecessary preprocessor concatenation. (I believe this was a relic of generalization that was removed earlier.)
* Style: Call function pointers without explicitly dereferencing. We were mostly consistent about this, with just a few exceptions.
* Cleanup `iosptrs.cpp` casting. We need to `reinterpret_cast` from function pointer to `void*` when calling `EncodePointer` and `DecodePointer` (doing so implicitly is a Microsoft bug-extension that Clang would warn about, if we used it to build the STL). Change C casts to `reinterpret_cast` for the return values. Mark `pf` as `const` and use `auto` for style.
* Cleanup `iosptrs.cpp` initialization. Change `= {0}` for the array of function pointers to `= {}` (to avoid `0`-as-null). Strangely, we were initializing a `size_t` with `= {_Nats};` which I've never seen elsewhere; change this to plain copy-init.
* Change `NULL` to `nullptr`.
* Change `0` to `nullptr`. This isn't attempting to be exhaustive (there are still many occurrences in the numeric conversion machinery).
* Change C casts to `reinterpret_cast` and don't cast away `const` unnecessarily. This works towards #184 but doesn't attempt to be exhaustive.

### CODE REVIEW

File: stl/src/awint.h [hunk 1]
```diff
@@ -294,57 +290,57 @@ enum wrapKERNEL32Functions {
...
+using PFNGETLOCALEINFOEX = int(WINAPI*)(LPCWSTR, LCTYPE, LPWSTR, int);
+using PFNLCMAPSTRINGEX   = int(WINAPI*)(LPCWSTR, DWORD, LPCWSTR, int, LPWSTR, int, LPNLSVERSIONINFO, LPVOID, LPARAM);
 
 // Use this macro for caching a function pointer from a DLL
 #define STOREFUNCTIONPOINTER(instance, functionname) \
-    __KERNEL32##Functions[e##functionname] = (PVOID)(GetProcAddress(instance, #functionname));
+    __KERNEL32Functions[e##functionname] = reinterpret_cast<PVOID>(GetProcAddress(instance, #functionname));
 
 // Use this macro as a cached function pointer from a DLL
 #define DYNAMICGETCACHEDFUNCTION(functiontypedef, functionname, functionpointer) \
```
Comment (CaseyCarter) [inline stl/src/awint.h:338]: Mega-nitpick: `functiontypedef` really should be `functiontype`.

Comment (StephanTLavavej) [reply stl/src/awint.h]: Giga-nitpick: This is a function pointer type, not a function type. :smiley_cat:

I've renamed this and the other macro parameters. Please meow if this needs further change.

File: stl/src/awint.h [hunk 2]
```diff
@@ -290,45 +290,45 @@ enum wrapKERNEL32Functions {
...
+using PFNACQUIRESRWLOCKEXCLUSIVE         = VOID(WINAPI*)(PSRWLOCK);
+using PFNTRYACQUIRESRWLOCKEXCLUSIVE      = BOOLEAN(WINAPI*)(PSRWLOCK);
+using PFNRELEASESRWLOCKEXCLUSIVE         = VOID(WINAPI*)(PSRWLOCK);
+using PFNSLEEPCONDITIONVARIABLESRW       = BOOL(WINAPI*)(PCONDITION_VARIABLE, PSRWLOCK, DWORD, ULONG);
+using PFNCREATETHREADPOOLWORK            = PTP_WORK(WINAPI*)(PTP_WORK_CALLBACK, PVOID, PTP_CALLBACK_ENVIRON);
+using PFNSUBMITTHREADPOOLWORK            = VOID(WINAPI*)(PTP_WORK);
+using PFNCLOSETHREADPOOLWORK             = VOID(WINAPI*)(PTP_WORK);
+using PFNCOMPARESTRINGEX = int(WINAPI*)(LPCWSTR, DWORD, LPCWSTR, int, LPCWSTR, int, LPNLSVERSIONINFO, LPVOID, LPARAM);
+using PFNGETLOCALEINFOEX = int(WINAPI*)(LPCWSTR, LCTYPE, LPWSTR, int);
+using PFNLCMAPSTRINGEX   = int(WINAPI*)(LPCWSTR, DWORD, LPCWSTR, int, LPWSTR, int, LPNLSVERSIONINFO, LPVOID, LPARAM);
```
Comment (mnatsuhara) [inline stl/src/awint.h:331]: Are we not lining up the `=` because it causes the line to exceed 120 characters?

Comment (CaseyCarter) [reply stl/src/awint.h:331]: Yes, that's almost certainly what clang-format is doing here.

Comment (StephanTLavavej) [reply stl/src/awint.h:331]: When clang-format works, it really generates nice results. Now if only it understood variable templates and concepts...

Comment (CaseyCarter) [reply stl/src/awint.h:331]: > When clang-format works, it really generates nice results.

Isn't this [tautological](https://xkcd.com/703/)?

Comment (CaseyCarter) [reply stl/src/awint.h:331]: Related: Should we have autolink support for XKCD-XXXX?

**Pull request merged**